vtk-m: update Vtk-m pkg to its latest version v1.5.1

vicentebolea opened the pull request:
Hello there!

Last month the __VTK-m__ team released its latest version named `v1.5.1`. This new
release only contains `bugfixes` related to compilers error and warnings.

This PR also: 
- Changes the dependency to depend on CMake >= 3.12
- Sets `VTKm_NO_ASSERT` to `ON` by default

Signed-off-by: Vicente Adolfo Bolea Sanchez <vicente.bolea@kitware.com>

vicentebolea committed:
vtk-m: update Vtk-m pkg to its latest version v1.5.1

Last month VTK-m releases its lastest version named `v1.5.1`. This new
release only contains bugfixes related to compiler error / warnings.

- Depends on CMake >= 3.12
- Set VTKm_NO_ASSERT=ON by default
- add maintainers

Signed-off-by: Vicente Adolfo Bolea Sanchez <vicente.bolea@kitware.com>

adamjstewart commented on var/spack/repos/builtin/packages/vtk-m/package.py:
```diff
@@ -44,7 +45,7 @@ class VtkM(CMakePackage, CudaPackage):
             description="enable 64 bits ids")
 
-    depends_on("cmake")
+    depends_on("cmake@3.12:")         # CMake >= 3.12
```
```suggestion
    depends_on("cmake@3.12:", type="build")         # CMake >= 3.12
```
I assume this dependency is only needed at build-time

adamjstewart reviewed (commented):
@vincentebolea are you or anyone else at Kitware interested in being the official "maintainer" of this Spack package, or any other Kitware software? If so, just add:
```python
maintainers = ['vincentebolea']
```
to those packages (can have multiple maintainers). You don't have to be a Spack expert, just someone willing to review PRs or take a look at build issues relating to these packages.

vicentebolea commented:
@adamjstewart Thanks for pointing out about the maintainers, I brought it today to the team and we designated 3 of us for the task :+1: 

Also thanks for the suggested change!

I added the changed and rebased it, let me if it needs more iterations.

adamjstewart commented:
Thanks @vicentebolea, if you want to add yourselves as maintainers for other any packages just submit a PR!
